tests/integration/CapabilityTest.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+namespace Automattic\LegacyRedirector\Tests;
+
+use Automattic\LegacyRedirector\Capability;
+
+/**
+ * CapabilityTest class
+ */
+class CapabilityTest extends TestCase {
+
+	/**
+	 * We remove the Capability::MANAGE_REDIRECTS_CAPABILITY  before testing
+	 *
+	 * @beforeClass
+	 */
+	public static function beforeClassSetup() {
+		$capability = new Capability();
+		$capability->unregister();
+	}
+
+	public function test_admin_capabilities_before_register() {
+		$user_id = $this->factory->user->create( array( 'role' => 'administrator' ) );
+		$user    = wp_set_current_user( $user_id );
```

Comment:
As as separate test, can you grab user ID 1 and check their caps before and after registering?

So rename this test to `test_new_admin_capabilities_before_register` and create a `test_existiing_admin_capabilities_before_register` test method?

Reply:
We'll also want to see tests for when `wpcom_vip_add_role_caps()` exists and when it doesn't since right now I suspect only one of those branches is being tested.

Reply:
I've added the ID=1 user test, but for multisite, this won't work because of in WP_User class, on multisite the has_cap [returns true](https://github.com/WordPress/WordPress/blob/master/wp-includes/class-wp-user.php#L754-L760) for admin users

Not sure about wpcom_vip_add_role_caps , we might need to mock it, but i think we should do that in a separate PR?